### PR TITLE
added timestamp.js

### PR DESCRIPTION
**Pull request by madguy02**

### CODE REVIEW

**Comment by Sing-Li** [comment]
A good start.   

We likely will deviate from the godoc for the APIs - due to threading model differences between nodeJS and Go affecting the interface.   This means we should think  about our own doc for the API  - or find tools that can generate API docs from code.

**Comment by Sing-Li** [comment]
@madguy02   this will need to become an npm module.  Can you please structure it to be a publishable npm module?

**Comment by madguy02** [comment]
@Sing-Li  cool i will do it

File: nodematrixserverlib/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "nodematrixserverlib",
+  "version": "1.0.0",
+  "description": "A library for common functions in nodejs",
+  "main": "timestamp.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (Sing-Li) [inline nodematrixserverlib/package.json]: Please add unit test, and the corresponding directories.   Include a trivial unit test for now, unless you can think of something meaningful for timestamps :)

File: nodematrixserverlib/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "nodematrixserverlib",
+  "version": "1.0.0",
```
Comment (Sing-Li) [inline nodematrixserverlib/package.json]: Common industry practice is to start with 0.1.0 and increment it as you build the lib.

You definitely don't want to start with 1.0.0 with almost no implementation.

File: nodematrixserverlib/package.json [hunk 3]
```diff
@@ -18,5 +18,11 @@
   "bugs": {
     "url": "https://github.com/RocketChat/Rocket.Chat.Federation/issues"
   },
-  "homepage": "https://github.com/RocketChat/Rocket.Chat.Federation#readme"
+  "homepage": "https://github.com/RocketChat/Rocket.Chat.Federation#readme",
+  "dependencies": {
+    "chai": "^3.5.0",
+    "express": "^4.15.2",
+    "mocha": "^3.3.0",
```
Comment (Sing-Li) [inline nodematrixserverlib/package.json]: @madguy02   Consider not including unit test framework(s) with production builds.

Take a look at the `devDependencies` section of package.json.

**Comment by madguy02** [comment]
hii @Sing-Li  please review this

**Comment by Sing-Li** [comment]
@madguy02 

(1) version number - semver, do not start with 1.0.0
(2) move tests framework away from `dependencies`

Thanks.

**Comment by madguy02** [comment]
cool  fixing it

**Review by Sing-Li — APPROVED** [review]

**Pull request merged**